src/providers.rs

```diff
@@ -46,361 +39,219 @@ pub const LLAMA_ARBITRUM_ONE_HOSTNAME: &str = "arbitrum.llamarpc.com";
-        ),
-        (
-            RpcService::EthSepolia(EthSepoliaService::BlockPi),
-            BLOCKPI_ETH_SEPOLIA_HOSTNAME,
-        ),
-        (
-            RpcService::EthSepolia(EthSepoliaService::PublicNode),
-            PUBLICNODE_ETH_SEPOLIA_HOSTNAME,
-        ),
-        (
-            RpcService::ArbitrumOne(L2MainnetService::Alchemy),
-            ALCHEMY_ARBITRUM_ONE_HOSTNAME,
-        ),
-        (
-            RpcService::ArbitrumOne(L2MainnetService::Ankr),
-            ANKR_HOSTNAME,
-        ),
-        (
-            RpcService::ArbitrumOne(L2MainnetService::BlockPi),
-            BLOCKPI_ARBITRUM_ONE_HOSTNAME,
-        ),
-        (
-            RpcService::ArbitrumOne(L2MainnetService::PublicNode),
-            PUBLICNODE_ARBITRUM_ONE_HOSTNAME,
-        ),
-        (
-            RpcService::BaseMainnet(L2MainnetService::Alchemy),
-            ALCHEMY_BASE_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::BaseMainnet(L2MainnetService::Ankr),
-            ANKR_HOSTNAME,
-        ),
-        (
-            RpcService::BaseMainnet(L2MainnetService::BlockPi),
-            BLOCKPI_BASE_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::BaseMainnet(L2MainnetService::PublicNode),
-            PUBLICNODE_BASE_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::OptimismMainnet(L2MainnetService::Alchemy),
-            ALCHEMY_OPT_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::OptimismMainnet(L2MainnetService::Ankr),
-            ANKR_HOSTNAME,
-        ),
-        (
-            RpcService::OptimismMainnet(L2MainnetService::BlockPi),
-            BLOCKPI_OPTIMISM_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::OptimismMainnet(L2MainnetService::PublicNode),
-            PUBLICNODE_OPTIMISM_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::EthMainnet(EthMainnetService::Llama),
-            LLAMA_ETH_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::ArbitrumOne(L2MainnetService::Llama),
-            LLAMA_ARBITRUM_ONE_HOSTNAME,
-        ),
-        (
-            RpcService::BaseMainnet(L2MainnetService::Llama),
-            LLAMA_BASE_MAINNET_HOSTNAME,
-        ),
-        (
-            RpcService::OptimismMainnet(L2MainnetService::Llama),
-            LLAMA_OPTIMISM_MAINNET_HOSTNAME,
-        ),
-    ]
+    pub static PROVIDER_MAP: HashMap<ProviderId, Provider> =
+        PROVIDERS.with(|providers| providers.iter()
+            .map(|provider| (provider.provider_id, provider.clone())).collect());
+
+    pub static SERVICE_PROVIDER_MAP: HashMap<RpcService, ProviderId> = hashmap! {
+        RpcService::EthMainnet(EthMainnetService::Cloudflare) => 0,
+        RpcService::EthMainnet(EthMainnetService::Ankr) => 1,
+        RpcService::EthMainnet(EthMainnetService::PublicNode) => 2,
+        RpcService::EthMainnet(EthMainnetService::BlockPi) => 3,
+        // RpcService::EthMainnet(EthSepoliaService::Sepolia) => 4,
```

Comment:
Do we want to remove this provider (https://rpc.sepolia.org) or introduce a corresponding `EthSepoliaService::Sepolia`?

Reply:
It's not clear to me why you deleted it? I would avoid any breaking change in the provider IDs since this is user facing. Can you give here more details for the reason of this change?

Reply:
This `rpc.sepolia.org` provider has always been in the canister without a corresponding Candid type, so the question here is whether we want to add `EthSepoliaService::Sepolia` for completeness (backwards-compatible change IIUC), remove the provider (breaking change), or exclude the provider from the validation logic (same as before).

Reply:
Thanks Ryan for the explanation! I would a priori add a corresponding Candid type. Was there a reason for avoiding a Candid type in the first place or was it just something that got forgotten?

Reply:
IIRC, this provider was outside of the agreed-upon list of Sepolia RPC providers in the design document, but it made sense to include the official Sepolia endpoint for those who wanted to use it. 

PR: https://github.com/dfinity/ic/pull/1113